cirq-core/cirq/protocols/has_unitary_protocol.py

```diff
@@ -114,6 +123,12 @@ def has_unitary(val: Any, *, allow_decompose: bool = True) -> bool:
     return False
 
 
+def _strat_has_measurement(val: Any) -> Optional[bool]:
+    if isinstance(val, GateOperation) and isinstance(val.gate, MeasurementGate):
```

Comment:
May as well check `PauliMeasurementGate` too.

Reply:
That sounds like a cleaner approach indeed. Since your comment seems to conflict my previous approach and others commented on it, I decided to address their comments first, commit, then undo my changes and implement your approach.

Done.